### DISCUSSION TITLE
BLE Data Length Extension doesn't work well

### DISCUSSION THREAD

**Discussion by kaiser-ren** [zephyrproject-rtos/zephyr#42779] [General]
**Describe the bug**
I used nRF52840dk as a peripheral which run ./example/bluetooth/peripheral_ots example, and my iOS device as central. I also used nRF Sniffer as packet tracer. 
I found that nRF52840dk did a LL_LENGTH_REQ after the connection established, but max tx octets and max rx octets are 27. I think the correct value should be 251 because nRF52840dk as a Bluetooth controller support Data Length Extension. 

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? nRF52840dk

**To Reproduce**
1. cd ./example/bluetooth/peripheral_ots
2. west build -p auto -b nrf52840dk_nrf52840 ./
3. west flash
4. Use "nRF Connect" app on iOS device to connect this DK. 
5. Use a packet sniffer, it can be nRF sniffer, Ellisys or Frontline to capture LL_LENGTH_REQ/LL_LENGTH_RSP

**Expected behavior**
The max tx octet and max rx octet of LL_LENGTH of nRF52840dk are 251

**Impact**
This impacts the performance of throughput of Zephyr BLE application. 

**Logs and console output**
[DLE works abnormal.pcapng.zip](https://github.com/zephyrproject-rtos/zephyr/files/8053784/DLE.works.abnormal.pcapng.zip)

**Comment by LingaoM** [comment]
https://github.com/nrfconnect/sdk-nrf/tree/main/samples/bluetooth/throughput

**Comment by carlescufi** [comment]
@cvinayak can you please comment on this thread?

**Comment by cvinayak** [comment]
@LingaoM Please check whether have the below in your prj.conf, if not, then only the default 27 byte PDUs will be used by the Controller.

```
CONFIG_BT_BUF_ACL_RX_SIZE=251
CONFIG_BT_L2CAP_TX_MTU=247
CONFIG_BT_BUF_ACL_TX_SIZE=251
CONFIG_BT_CTLR_DATA_LENGTH_MAX=251
```

**Comment by DuMaM** [comment]
I'm facing similar issue with nrf sdk 2.0.2, nrf5340 DK and this particular example.

**Comment by cvinayak** [comment]
Please check if your hci_rpmsg sample has the above mentioned Kconfig values.

**Reply by DuMaM** [reply]
Here are my configs
https://github.com/DuMaM/bitly_nrf5x/blob/feature/add-dumping-meas/prj_nrf5340dk_nrf5340_cpuapp.conf
https://github.com/DuMaM/bitly_nrf5x/blob/feature/add-dumping-meas/child_image/hci_rpmsg.conf

**Reply by cvinayak** [reply]
As you are using the nRF Connect SDK, you have to contact https://devzone.nordicsemi.com/

Did you try using the vanilla zephyrproject-rtos, building the hci_rpmsg sample with the said Kconfig?

**Reply by DuMaM** [reply]
> As you are using the nRF Connect SDK, you have to contact https://devzone.nordicsemi.com/

Ok I will do

> Did you try using the vanilla zephyrproject-rtos, building the hci_rpmsg sample with the said Kconfig?

Not yet, but thats a good idea.

Strangely it's working fine for Particle Xenon Board with nrf52840
